openapi.yml

```diff
@@ -77,18 +115,17 @@ components:
           $ref: "#/components/schemas/TokenAmount"
         sellToken:
           $ref: "#/components/schemas/Address"
-        sellTokenTip:
+        tip:
           $ref: "#/components/schemas/TokenAmount"
         orderType:
-          description: Is this a buy order or sell order?
-          type: string
-          enum: [buy, sell]
+          $ref: "#/components/schemas/OrderType"
         fillType:
           description: Is this a fill-or-kill order a partially fillable order?
           type: string
           enum: [fillorkill, partial]
-        nonce:
-          $ref: "#/components/schemas/Nonce"
+        appData:
+          description: Identifier to make order unique or pass in arbitrary data. uint32 encoded in decimal.
+          type: integer
```

Comment:
@anxolin 
`validTo` has to be JSON stringifiable, so either number (for epoch) or string
Do you mean something like `JSON.stringify(new Date()) > ""2020-11-25T07:01:35.939Z""`?

Reply:
It could be a date too, but this is very specific to how it is interpreted in the smart contract which is the number of seconds since some epoch and it is part of the signed data so we have to make sure that any date conversions don't lose information.

Reply:
Yes, I understand. But why a ISO date would lose information?
It would make it easier for user to submit using their time zone, or UTC if they prefer so. 

Otherwise, I feel we expose implementation details.

Reply:
>which is the number of seconds since some epoch

It will even be the seconds since the unix time counting. Federico checked that even in this case a uint32 would be sufficient for many upcoming years. ( I updated that descriptions)

Hence, I think it is fine to transfer the validTo information by a commonly used unix timestamp and it is not to specific to our implementation

Reply:
Personally, I also like @anxolin's suggestion from an API perspective - it makes sense for the service to convert this timestamp to whatever representation it needs.

That being said, because we already require a user signature for the order, it already needs to be converted to the appropriate integer format (Unix timestamp, or seconds from Unix epoch). So with that in mind maybe it makes most sense to just use the `validTo` as an integer so that the data being signed is the same as the data being submitted with the order?

Reply:
👌 That makes total sense. it's a very low level endpoint, so it's good to send what you sign